doc/user_guide/interactions.rst

```diff
@@ -695,12 +704,148 @@ often provides a more convenient syntax
+    color_europe = alt.param(value="#ffb54d", bind=alt.binding(input='color', name='Europe '))
+    color_japan = alt.param(value="#adadad", bind=alt.binding(input='color', name='Japan '))
+
+    alt.Chart(data.cars.url).mark_circle().encode(
+        x='Miles_per_Gallon:Q',
+        y='Horsepower:Q',
+        color=alt.Color(
+            'Origin:N',
+            scale=alt.Scale(
+                domain=['USA', 'Europe', 'Japan'],
+                range=[color_usa, color_europe, color_japan]
+            )
+        )
+    ).add_params(
+        color_usa, color_europe, color_japan
+    )
+
+Encoding Channel Binding
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+There is no direct way to map an encoding channel to a widget in order to dynamically display different charts based on different column choices, such as ``y=column_param``. The underlying reason this is not possible is that in Vega-Lite, the ``field`` property does not accept a parameter as value; see the `field Vega-Lite documentation <https://vega.github.io/vega-lite/docs/field.html>`_. You can follow the discussion in this issue https://github.com/vega/vega-lite/issues/7365, and in the meantime, you can use parameters for a convenient workaround which let's you achieve the same functionality and change the plotted columns based on a widget selection (the x-axis title cannot be changed dynamically, but a text mark could be used instead if desired):
+
+.. altair-plot::
+
+    dropdown = alt.binding_select(
+        options=['Miles_per_Gallon', 'Displacement', 'Weight_in_lbs', 'Acceleration'],
+        name='X-axis column '
+    )
+    xcol_param = alt.param(
+        value='Miles_per_Gallon',
+        bind=dropdown
+    )
+
+    alt.Chart(data.cars.url).mark_circle().encode(
+        x=alt.X('x:Q', title=''),
+        y='Horsepower:Q',
+        color='Origin:N'
+    ).transform_calculate(
+        x=f'datum[{xcol_param.name}]'
+    ).add_params(
+        xcol_param
+    )
+
+It was possible to achieve something similar before the introduction of parameters in Altair 5, but the spec for this is more involved so the solution above is to prefer:
+
+.. altair-plot::
+
+    # Similar workaround for earlier versions of Altair
+    source = data.cars().melt(id_vars=['Origin', 'Name', 'Year', 'Horsepower', 'Cylinders'])
+    dropdown_options = source['variable'].drop_duplicates().tolist()
+
+    dropdown = alt.binding_select(
+        options=dropdown_options,
+        name='X-axis column'
+    )
+    selection = alt.selection_point(
+        fields=['variable'],
+        value=[{'variable': dropdown_options[0]}],
+       # init={'variable': dropdown_options[0]},  # For Altair 4
+        bind=dropdown
+    )
+
+    alt.Chart(source).mark_circle().encode(
+        x=alt.X('value:Q', title=''),
+        y='Horsepower',
+        color='Origin',
+    ).add_params(
+        selection
+    ).transform_filter(
+        selection
+    )
```

Comment:
```suggestion
```
I think we can delete this part. Just stick to the new routine of altair v5. This page is so extensive already.